en/ui-system/components/engine/ui-batch.md

```diff
@@ -1,19 +1,90 @@
+
+1. Property **BATCHER2D_MEM_INCREMENT** under **Project Properties -> Macro Configuration** indicates the maximum vertex number for a MeshBuffer. Increasing the value will allow a MeshBuffer to host more data to be rendered but will also increase memory consumption.
+
+2. **BATCHER2D_MEM_INCREMENT** is measured in **kilobytes**. Users can follow the instructions below to calculate the corresponding capacity for vertex numbers:
+
+    Vertices in Cocos Creators are formatted as [vfmtPosUvColor](https://github.com/cocos-creator/engine/blob/v3.4.1/cocos/2d/renderer/vertex-format.ts#L43), its definitions as follows:
+
+    ```ts
+    export const vfmtPosUvColor = [
+    // RGB32F represents three 32-bit float numbers.
+    new Attribute(AttributeName.ATTR_POSITION, Format.RGB32F),
+    // RG32F represents two 32-bit float numbers.
+    new Attribute(AttributeName.ATTR_TEX_COORD, Format.RG32F),
+    // RGBA32F represents four 32-bit float numbers.
+    new Attribute(AttributeName.ATTR_COLOR, Format.RGBA32F),
+    ];
+    ```
+
+    It can be inferred that a vertex takes 9 `float` numbers, which in memory space equates to: `1 * 9 * 4 / 1024 (KB)`, specifically:
+
+      - **1** is the number of vertices
+
+      - **9** is the number of `floats` defined in each `vfmtPosUvColor`
+
+      - **4** is the bytes occupied by each `float`
+
+      - **1024** is the transference rate from bytes to kilobytes
+
+    The default value for **BATCHER2D_MEM_INCREMENT** is 144KB with a capacity of `144 * 1024 / (9 * 4）= 4096` vertices.
+    
+    Please note that the maximum capacity for a MeshBuffer is **65535**. In other words, the maximum value for **BATCHER2D_MEM_INCREMENT** is `65535 * 9 * 4 / 1024 ≈ 2303.96` kilobytes.
+
+3. The core mechanism for 2D rendering is **static vertex buffer**, specifically:
+
+    - **Static vertex buffer** presides over the whole lifespan of components while index buffer is filled on a per frame basis and does not implement memory caches in any form. Due to the volume of data in vertex buffer, it is advisable to store it separately to give access to specific memory segments while avoiding unnecessary updates.
+
+    - Index buffer has a simpler data structure and lighter data volume compared to vertex buffer. Updating at each frame, index buffer defines the order for vertices to be rendered and takes an insignificant amount of processing power, requiring no sophisticated memory management.
+
+    - Due to vertex buffer being static, it is advisable to preload vertex buffer at the very beginning of the component's lifespan. At loading, component will request relevant vertex buffers from MeshBuffer and returns them at destruction.
+
+    - When MeshBuffer is unable to provide the vertex buffer requested by the component, the engine will create a new MeshBuffer allocated with the amount of memory space as dictated in **BATCHER2D_MEM_INCREMENT** so that vertex buffer can be successfully distributed.
```

Comment:
```suggestion
    - When MeshBuffer is unable to provide the vertex buffer requested by the component, the engine will create a new MeshBuffer allocated with the amount of memory space as dictated in **BATCHER2D_MEM_INCREMENT** so that vertex buffer can be successfully distributed.

```